src/NuGet.Core/NuGet.Commands/SourcesCommands/SourcesRunner.cs

```diff
@@ -0,0 +1,299 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Linq;
+using NuGet.Common;
+
+namespace NuGet.Commands
+{
+    /// <summary>
+    /// Shared code to run the "sources" command from the command line projects
+    /// </summary>
+    public static class SourcesRunner
+    {
+        public static void Run(SourcesArgs args)
+        {
+            switch (args.Action)
+            {
+                case SourcesAction.List:
+                    switch (args.Format)
+                    {
+                        case SourcesListFormat.Short:
+                            PrintRegisteredSourcesShort(args);
+                            break;
+                        case SourcesListFormat.Detailed:
+                        default:
+                            PrintRegisteredSourcesDetailed(args);
+                            break;
+                    }
+                    break;
+                case SourcesAction.Add:
+                    AddNewSource(args);
+                    break;
+                case SourcesAction.Remove:
+                    RemoveSource(args);
+                    break;
+                case SourcesAction.Enable:
+                    EnableOrDisableSource(args, enabled: true);
+                    break;
+                case SourcesAction.Disable:
+                    EnableOrDisableSource(args, enabled: false);
+                    break;
+                case SourcesAction.Update:
+                    UpdatePackageSource(args);
+                    break;
+            }
+        }
+
+        private static void EnableOrDisableSource(SourcesArgs args, bool enabled)
+        {
+            if (string.IsNullOrEmpty(args.Name))
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                    Strings.SourcesCommandNameRequired));
+            }
```

Comment:
args are validated in almost all the methods. Some refactoring is required to avoid redundancy.

Reply:
boilerplate reply: the code in SourcesRunner.cs was moved from SourcesCommand.cs - I have tried to make the minimal changes to it. That said, I'll review each of these comments. thx.

Reply:
gonna leave this for now...as different actions require different sets of arguments.